src/modules/collections/controllers/collections.controller.ts

```diff
@@ -106,10 +106,39 @@ export class CollectionsController {
 		const visitorSpaceAccessInfo =
 			await this.ieObjectsService.getVisitorSpaceAccessInfoFromUser(user);
 
+		const isKeyUser = user.getIsKeyUser();
+
 		// Limit access to the objects in the collection
-		folderObjects.items = (folderObjects.items ?? []).map((object) => {
-			return this.ieObjectsService.limitObjectInFolder(object, user, visitorSpaceAccessInfo);
-		});
+		folderObjects.items = (folderObjects.items ?? [])
+			.filter((object) => {
+				// if user is no keyUser AND object has ONLY license INTRA_CP-CONTENT AND/OR INTRA_CP-METADATA_ALL, do not show object
+				const noKeyUserAndHasOnlyLicense =
+					!isKeyUser &&
+					object.licenses &&
+					object.licenses.length === 1 &&
+					object.licenses.includes(
+						IeObjectLicense.INTRA_CP_CONTENT || IeObjectLicense.INTRA_CP_METADATA_ALL
+					);
+				const noKeyUserAndHasLicenses =
+					!isKeyUser &&
+					object.licenses &&
+					object.licenses.length === 2 &&
+					object.licenses.includes(
+						IeObjectLicense.INTRA_CP_CONTENT && IeObjectLicense.INTRA_CP_METADATA_ALL
+					);
+				if (noKeyUserAndHasOnlyLicense || noKeyUserAndHasLicenses) {
+					return;
+				} else {
+					return object;
+				}
```

Comment:
i think this does the same thing:
Also best to always return only booleans in a filter function, not undefined or objects
```suggestion
				const hasOnlyIntraCpLicenses =
					(object.licenses || []).filter(license => 
						![IeObjectLicense.INTRA_CP_CONTENT, IeObjectLicense.INTRA_CP_METADATA_ALL].includes(license)
					).length === 0;

				if (!isKeyUser && hasOnlyIntraCpLicenses) {
					return false;
				} else {
					return true;
				}
```